src/game/counting.rs

```diff
@@ -0,0 +1,276 @@
+            let mut possible_run = all_cards.as_slice()[i..].to_vec();
+            possible_run.sort();
+            if let Some(c) = score_run(possible_run) {
+                score.combinations.push(c);
+                break; // stop on the largest run
+            }
+        }
+    }
+
+    score.total_score = score.combinations.iter().map(|combi| combi.points).sum();
+    Ok(score)
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::game::cards::{Card, Rank::*, Suit as Of};
+    use card as c;
+
+    macro_rules! test_case {
+        ($name:ident,$cards_played:expr,$card:expr,$expected_score:literal, $expect_error:literal) => {
+            #[test]
+            fn $name() {
+                let computer_score = super::score_counting_cards_played($cards_played, $card);
+
+                match computer_score {
+                    Ok(s) => {
+                        assert_eq!(
+                            $expected_score, s.total_score,
+                            "Player Algo Score: {} vs. Hand Score: {}",
+                            s.total_score, $expected_score
+                        );
+                    }
+                    Err(e) => {
+                        if $expect_error == false {
+                            assert_eq!(true, false, "Error in scoring: {:?}", e);
+                        }
+                    }
+                }
+            }
+        };
+    }
+    test_case!(counting_first_card, &[], c!(Five, Of::Diamonds), 0, false);
+    test_case!(
+        counting_2_fives,
+        &[c!(Five, Of::Hearts),],
+        c!(Five, Of::Diamonds),
+        2,
+        false
+    );
+    test_case!(
+        counting_3_fives,
+        &[c!(Five, Of::Hearts), c!(Five, Of::Clubs),],
+        c!(Five, Of::Diamonds),
+        8,
+        false
+    );
+    test_case!(
+        counting_4_fives,
+        &[
+            c!(Five, Of::Hearts),
+            c!(Five, Of::Clubs),
+            c!(Five, Of::Diamonds),
+        ],
+        c!(Five, Of::Spades),
+        12,
+        false
+    );
+    test_case!(
+        counting_no_points,
+        &[
+            c!(Five, Of::Hearts),
+            c!(Five, Of::Clubs),
+            c!(Five, Of::Diamonds),
+            c!(Five, Of::Spades),
+        ],
+        c!(Ten, Of::Spades),
+        0,
+        false
+    );
+    test_case!(
+        counting_expect_error,
+        &[
+            c!(Five, Of::Hearts),
+            c!(Five, Of::Clubs),
+            c!(Five, Of::Diamonds),
+            c!(Five, Of::Spades),
+            c!(Ten, Of::Spades),
+        ],
+        c!(Ten, Of::Clubs),
+        0,
+        true
+    );
+    test_case!(
+        counting_31,
+        &[
+            c!(Five, Of::Hearts),
+            c!(Five, Of::Clubs),
+            c!(Five, Of::Diamonds),
+            c!(Five, Of::Spades),
+            c!(Ten, Of::Spades),
+        ],
+        c!(Ace, Of::Clubs),
+        2,
+        false
+    );
+
```

Comment:
Is there a method to the empty line here and not in other places?

Reply:
mostly accidental, but I do find it easer to read when there is whitespace there -- consistency is important.  no whitepace or whitespace?

Reply:
i wish the auto-formatter fixed this!

Reply:
In terms of consistency, I prefer the empty line between tests.